pkg/apis/provisioning/v1alpha5/provisioner_status.go

```diff
@@ -29,6 +30,9 @@ type ProvisionerStatus struct {
 	// its target, and indicates whether or not those conditions are met.
 	// +optional
 	Conditions apis.Conditions `json:"conditions,omitempty"`
+
+	// Resources is the list of resources that have been provisioned.
+	Resources v1.ResourceList `json:"resources,omitempty"`
```

Comment:
Nice call. I think there's definite value in having that in ProvisionerSpec. Might even help show how much better the instance utilization is with Karpenter. I think in the API though, it should could maybe be Provisioner.Status.Requests? 

Agreed though, that it might mean we update the provisioner on each pod event so I'm a little concerned from a scaling front.

Reply:
Controller-runtime will batch the reconciles and has a serial execution guarantee. I think this will roughly translate into a linear stream of writes to the API Server as it detects new resources. AFAICT, it shouldn't be enough to overwhelm anything.